Indexing/InductionFormulaIndex.hpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * This file is part of the source code of the software program
+ * Vampire. It is protected by applicable
+ * copyright laws.
+ *
+ * This source code is distributed under the licence found here
+ * https://vprover.github.io/license.html
+ * and in the source directory
+ */
+/**
+ * @file InductionFormulaIndex.hpp
+ * Defines class InductionFormulaIndex.
+ */
+
+
+#ifndef __InductionFormulaIndex__
+#define __InductionFormulaIndex__
+
+#include "Forwards.hpp"
+
+#include "Lib/DHMap.hpp"
+#include "Lib/Hash.hpp"
+#include "Lib/Stack.hpp"
+
+#include "Kernel/Clause.hpp"
+#include "Kernel/Substitution.hpp"
+
+namespace Inferences {
+  struct InductionContext;
+}
+
+namespace Indexing {
+
+using namespace Lib;
+using namespace Kernel;
+
+struct SecondaryStackHash {
```

Comment:
I still have some issues understanding how this should be implemented, so I'm open to suggestions if you think this would work better some other way. In particular, do I understand correctly that a collision (when this is called) is expected between two different keys? Is it enough to have a shallow hash like this?

I'm not really into exotic indexing so hopefully this remains a one timer.

Reply:
The secondary hash is only called if there is a collision on the first hash, you're right. Its only job is to return some integer reasonably quickly. Ideally it would be different for different keys that have collided on the first hash, but it's not compulsory.

So yes, this one is good ("sum of all the child stack lengths", right?) - but looking closer, starting with `FNV32_OFFSET_BASIS` is unnecessary, `unsigned res = 0;` would be fine here. Otherwise great!